src/Backend/Modules/Settings/Actions/Seo.php

```diff
@@ -60,6 +117,25 @@ private function validateForm(): void
                     $this->form->getField('seo_nofollow_in_comments')->getValue()
                 );
 
+                if ($this->isMultiLangual) {
+                    // save the linked pages for each language
+                    foreach ($this->workingLanguagePages as $pageId => $pageTitle) {
+                        $pageData = BackendPagesModel::get($pageId);
+                        $update['id'] = (int)$pageData['id'];
+                        $update['revision_id'] = (int)$pageData['revision_id'];
+                        $update['data'] = $pageData['data'];
+
+                        foreach ($this->languages as $lang => $language) {
+                            if ($lang != BL::getWorkingLanguage()) {
+                                $update['data']['hreflang_' . $lang] = $this->form->getField('page_' . $lang . '_' . $pageId)->getValue();
+                            }
+                        }
+
+                        $update['data'] = serialize($update['data']);
+                        BackendPagesModel::updateByIdAndRevisionId($update);
```

Comment:
Why not save it as a new revision?

Reply:
No specific reason, seems not necessary because the actual content of the page is not changed?

Reply:
I get that logic, but I'm also thinking about needing to port this to the restructure core branch where the pages module is running on doctrine already, and I think adding an additional way of updating pages adds complexity that can potentially cause issues in the future

Reply:
Aha, got it. It's quickly changed to a new revision. Let me know ;)

Reply:
I would just use the normal update method, that way it should be able to be ported without issues to the restructure-core branch

Reply:
Great, I will change that this week ;)

Reply:
Hmz, we need to build the whole page array to create a new revision, so also a new meta record.  What is the best way to create a new Meta record?
Building the array (title, keywords,...), add it manually to the db, and store the returned meta ID to the new revision?

Reply:
Hmm, maybe then make the scope smaller like update revisionData or something ?

Reply:
Ok, updated the function and changed the parameters so it's more clear this is only to update the data column